code/lib/csf-plugin/src/index.ts

```diff
@@ -1,13 +1,16 @@
+import { fileURLToPath } from 'node:url';
+
 import type { EnrichCsfOptions } from 'storybook/internal/csf-tools';
 
+import type { UnpluginFactory } from 'unplugin';
 import { createUnplugin } from 'unplugin';
 
 import { STORIES_REGEX } from './constants';
 import { rollupBasedPlugin } from './rollup-based-plugin';
 
 export type CsfPluginOptions = EnrichCsfOptions;
 
-export const unplugin = createUnplugin<CsfPluginOptions>((options) => {
+export const unpluginFactory: UnpluginFactory<EnrichCsfOptions> = (options) => {
```

Comment:
why was this refactor with necessary, splitting up the creation from the definition? I'm not seeing `unpluginFactory` being imported anywhere.

Reply:
I'm following the guide set by `unplugin`.

I upgraded in an attempt to try to avoid adding the peerDeps, I thought that perhaps the type problem would be solved.
I was not solved, so I had to add the peerDeps anyway, but I figured it be wasteful to undo the upgrade.
If it works, might as well keep it, right?

`unpluginFactory` isn't used, correct.
I could remove the export keyword, if you prefer.

Reply:
I see now. Looks like the `unpluginFactory` is only used in the docs/examples because it's being re-used by other modules in the package. Let's remove the export